agent/src/util/config_action.rs

```diff
@@ -276,6 +291,24 @@ async fn handle_config_delete(
     Ok(())
 }
 
+/// Checks to see if the configuration needs to be recreated.
+/// At present, this just checks to see if the `.metadata.generation` has changed.
+/// The `.metadata.generation` value is incremented for all changes, except for changes to `.metadata` or `.status`.
+async fn should_recreate_config(config: &Configuration, config_map: ConfigMap) -> Result<bool, ()> {
+    let name = config.metadata.name.clone().unwrap();
+    let last_generation = config_map.lock().await.get(&name).unwrap().last_generation;
+    trace!(
```

Comment:
I wonder if it is better to log this along with the result of the function once at the end of the function or maybe in the caller after line 159

Reply:
Yea...TBH we may not even need it. The logging we apply have up in the caller method is probably sufficient. The main reason I put it here, is that it kind of felt like we in general had at least trace logging in most methods to trace that we went into a method.